### PR TITLE
Add docs site to /docs directory on master branch

### PR DESCRIPTION
**Pull request by anselmbradford**
Fixes https://github.com/cfpb/capital-framework/issues/597

This is a first step in getting rid of the gh-pages branch hosted docs site, which is pain point for testing things and keeping the docs in sync with the changes in the codebase. This PR creates a `/docs` directory on the master branch and adds initial gulp tasks for handling the docs assets.

This PR adds a /docs directory with jekyll template content largely from gh-pages, with the following differences:
- Replaces homepage icons with SVGs.
- Fixes issue where homepage images had their `alt` attributes erroneously named `class`.
- Does not include `/components/grunt-tasks.md`.
- Does not include `src/static/css/cf-enhancements.less`
- Does not include `src/static/css/syntax.less` (appears to be redundant to `code.less`).
- Does not include `src/static/gulpfile.js`
- Does not include `src/static/js/main.js` and instead uses compiled `capital-framework.min.js`, though it probably should use a `main.js` script in the future.

## Additions

- Adds /docs directory with jekyll template content largely from gh-pages.
- Adds `gulp docs` task for copying documentation assets.
- Adds `gulp styles:docs` task for compiling documentation less files.

## Testing

1. `gulp docs` should pass.

If you want to run this locally you can follow the instructions on https://help.github.com/en/articles/setting-up-your-github-pages-site-locally-with-jekyll
and run jekyll inside the `/docs` directory. However, you have to remove `{{ site.github.url }}` from the templates (`default.html` and `home.html`) to get the styles to apply.

## Notes

- I was unclear if `_layouts/post.html` was used anywhere, but left it for now.

## Todos

- To actually host the docs this way we should probably merge this PR and then adjust the settings to point to it. See https://help.github.com/en/articles/configuring-a-publishing-source-for-github-pages#publishing-your-github-pages-site-from-a-docs-folder-on-your-master-branch

### CODE REVIEW

**Comment by Scotchester** [comment]
> * Replaces homepage icons with SVGs.

Were you working off an old version of gh-pages? [I did this a little while ago.](https://github.com/cfpb/capital-framework/commit/3ad33d29b7cf6d1917ba36e4915c05da0abfc029#diff-73caef5600fdbd8f2f74c3bc47f2c843) (including the external link icon in the footer)

> * Fixes issue where homepage images had their `alt` attributes erroneously named `class`.

It would be better to just empty out these alt attributes, since the images do not have any content value to screen reader users.

File: .gitignore [hunk 1]
```diff
@@ -66,9 +66,13 @@ vendor/
 static/vendor/
 src/vendor/
 util.js
-dist/
+/dist/
```
Comment (Scotchester) [inline .gitignore]:
```suggestion
/dist/  # docs/dist/ still checked in so GitHub Pages can use it to serve built static assets
```

Comment (anselmbradford) [reply .gitignore]: Shortened this to `# docs/dist/ checked in so GitHub Pages can serve built static assets.` to fit within 80 characters.

File: scripts/gulp/docs.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * Move documentation files into /docs.
+ * @returns {ChildProcess} A spawned process.
+ */
+function docs() {
+  const streams = [ _copyUsageDocs(), _copyCFJs(), _copyIcons() ];
+  return mergeStream( ...streams );
+}
+
+gulp.task( 'docs', docs );
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply scripts/gulp/docs.js]: Each task file should only encapsulate its own tasks, but we can put one like `build` in the gulpfile. I'll do this.

[the remark this answers is not in the document]
Comment (anselmbradford) [reply scripts/gulp/docs.js]: Done https://github.com/cfpb/capital-framework/pull/908/commits/99df0032c4ed0f9ed48d182ddc0f38b4fc113cce

File: docs/about/index.md [hunk 3]
```diff
@@ -3,23 +3,35 @@ layout: default
...
+by the Consumer Financial Protection Bureau with the following goals:
+consolidate front-end code patterns across CFPB web products
+in a collaborative way, adhere to design and UX standards set forth
+by the CFPB Design Manual, respect the guiding principles of accessibility,
+and support a wide range of browsers.
 
-As a work of the United States Government, all code is open source and in the public domain (excluding any exceptions listed within a Capital Framework component’s TERMS). We encourage you to use this framework in your own projects and to contribute back.
+As a work of the United States government, all code is open source and
+in the public domain (excluding any exceptions listed within in the project's
+[`TERMS.md` file](https://github.com/cfpb/capital-framework/blob/canary/TERMS.md#exceptions)).
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply docs/about/index.md]: https://github.com/cfpb/capital-framework/pull/908/commits/5af96b34c507771abd738f7ae6bc40470bddf66c

File: scripts/gulp/docs.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+function _copyUsageDocs() {
+  return gulp.src( './packages/**/usage.md' )
+    .pipe( gulp.dest( './docs/_includes/usage/' ) );
+}
+
+/**
+ * Copies the compiled Capital Framework (CF) JS into the doc site structure.
+ * TODO: This is different from how the CSS is handled,
+ * which includes CF source less files and compiles the docs CSS.
+ * The JavaScript should probably work this way to, to ensure that CF
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply scripts/gulp/docs.js]: https://github.com/cfpb/capital-framework/pull/908/commits/5af96b34c507771abd738f7ae6bc40470bddf66c

File: .gitignore (hunk 1 above)
Comment (anselmbradford) [reply .gitignore]: https://github.com/cfpb/capital-framework/pull/908/commits/5af96b34c507771abd738f7ae6bc40470bddf66c

File: docs/index.html [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+      <p class="sans">Capital Framework has been written to work with a wide range of devices and browsers. Following a modern, mobile first responsive approach, sites easily adapt to wide range of screen sizes all while carefully following accessibility best practices.</p>
+  </div>
+  <div class="content-l_col content-l_col-1-3">
+    <div class="section-image">
+      <img src="dist/img/project.png" alt="">
+    </div>
+      <h3>A solid project starter</h3>
+
+      <p class="sans">Setting up the framework for your project and creating new components should be quick and easy. By using our <a href="http://yeoman.io/">Yeoman</a> generator, you not only get access to a collection of <a href="http://lesscss.org/">Less</a>, CSS, and JavaScript files, but also a modern front end build process.</p>
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply docs/index.html]: https://github.com/cfpb/capital-framework/pull/908/commits/5af96b34c507771abd738f7ae6bc40470bddf66c

File: scripts/gulp/docs.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+ */
+function _copyUsageDocs() {
+  return gulp.src( './packages/**/usage.md' )
+    .pipe( gulp.dest( './docs/_includes/usage/' ) );
+}
+
+/**
+ * Copies the compiled Capital Framework (CF) JS into the doc site structure.
+ * TODO: This is different from how the CSS is handled,
+ * which includes CF source less files and compiles the docs CSS.
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply scripts/gulp/docs.js]: https://github.com/cfpb/capital-framework/pull/908/commits/5af96b34c507771abd738f7ae6bc40470bddf66c

**Comment by Scotchester** [comment]
Let's give it a whirl!

**Comment by Scotchester** [comment]
One other TODO we should state explicitly: If this works as expected, we'll need to update the contributing doc.

**Pull request merged**